user/tests/wiring/no_fixture/interrupts.cpp

```diff
@@ -0,0 +1,37 @@
+
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+test(interrupts_atomic_section)
+{
+	// using SysTick for convenience - we could use a stm32 timer interrupt for a more generic test
+	// millis() will not increment while interrupts are disabled.
+	// micros() does increment
+
+	uint32_t start_micros = micros();
+	uint32_t start_millis, end_millis;
+	{
+		ATOMIC_SECTION();
+		start_millis = millis();
+		while (micros()-start_micros<2000)
+		{
+			// validates that atomic sections can be nested and interrupts restored when the outermost one exits
+			ATOMIC_SECTION();
```

Comment:
The atomic section will disable interrupts until the end of the section (line 22), so millis() will not increment at all.  The block of code after tries it without atomic and millis() continues incrementing.